drivers/adc/Kconfig.mcux

```diff
@@ -5,36 +5,34 @@
 
 config ADC_MCUX_ADC12
 	bool "MCUX ADC12 driver"
-	depends on HAS_MCUX_ADC12
+	default y
+	depends on DT_HAS_NXP_KINETIS_ADC12_ENABLED
 	select PINCTRL
 	help
 	  Enable the MCUX ADC12 driver.
 
 config ADC_MCUX_ADC16
 	bool "MCUX ADC16 driver"
-	depends on HAS_MCUX_ADC16
+	default y
+	depends on DT_HAS_NXP_KINETIS_ADC16_ENABLED
 	select PINCTRL
 	help
 	  Enable the MCUX ADC16 driver.
 
 config ADC_MCUX_12B1MSPS_SAR
 	bool "MCUX 12B1MSPS SAR ADC driver"
-	depends on HAS_MCUX_12B1MSPS_SAR
+	default y
+	depends on DT_HAS_NXP_MCUX_12B1MSPS_SAR_ENABLED
 	help
 	  Enable the MCUX 12B1MSPS SAR ADC driver.
 
 config ADC_MCUX_LPADC
 	bool "MCUX LPADC driver"
-	depends on HAS_MCUX_LPADC
+	default y
+	depends on DT_HAS_NXP_LPC_LPADC_ENABLED
 	help
 	  Enable the MCUX LPADC driver.
 
-config ADC_MCUX_ETC
-	bool "MCUX ADC ETC driver"
-	depends on HAS_MCUX_ADC_ETC
-	help
-	  Enable the MCUX ADC ETC driver.
-
```

Comment:
Are you sure this can be removed? It was added in c343da51612dbdc40902a29e5898846eb91effce.

Reply:
cc @danieldegrasse @mmahadevan108

Reply:
This symbol doesn't have any driver and currently it's only used to enable `adc_etc` driver from the nxp_hal manually so it must stay as it is.